ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template3.kt

```diff
@@ -0,0 +1,199 @@
+import com.revenuecat.purchases.ui.revenuecatui.data.TestData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.selectedLocalization
+import com.revenuecat.purchases.ui.revenuecatui.helpers.isInPreviewMode
+
+private object Template3UIConstants {
+    val iconCornerRadius = 8.dp
+    val iconSize = 65.dp
+    val featureIconSize = 35.dp
+    val iconPadding = 5.dp
+}
+
+@Composable
+internal fun Template3(
+    state: PaywallViewState.Loaded,
+    viewModel: PaywallViewModel,
+) {
+    Column(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        Column(
+            modifier = Modifier
+                .weight(1f)
+                .fillMaxWidth()
+                .padding(
+                    horizontal = UIConstant.defaultHorizontalPadding,
+                    vertical = UIConstant.defaultVerticalSpacing,
+                ),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(UIConstant.defaultVerticalSpacing, Alignment.CenterVertically),
+        ) {
+            Template3MainContent(state, viewModel)
+        }
+        PurchaseButton(state, viewModel)
+        Footer(templateConfiguration = state.templateConfiguration, viewModel = viewModel)
+    }
+}
+
+@Composable
+private fun Template3MainContent(state: PaywallViewState.Loaded) {
+    IconImage(state.templateConfiguration.images.iconUri)
+    val localizedConfig = state.selectedLocalization
+    val colors = state.templateConfiguration.getCurrentColors()
+    Text(
+        style = MaterialTheme.typography.headlineSmall,
+        fontWeight = FontWeight.Black,
+        textAlign = TextAlign.Center,
+        text = localizedConfig.title,
+        color = colors.text1,
+    )
+    Spacer(modifier = Modifier.height(UIConstant.defaultVerticalSpacing * 2))
+    Features(
+        features = localizedConfig.features,
+        colors = colors,
+    )
+}
+
+@Composable
+private fun Features(
+    features: List<PaywallData.LocalizedConfiguration.Feature>,
+    colors: TemplateConfiguration.Colors,
+) {
+    if (features.isEmpty()) return
+
+    Column(
+        modifier = Modifier.verticalScroll(rememberScrollState())
+            .fillMaxHeight(),
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultVerticalSpacing, Alignment.Top),
+    ) {
+        features.forEach { feature ->
+            Feature(
+                feature = feature,
+                colors = colors,
+            )
+        }
+    }
+}
+
+@Composable
+private fun Feature(
+    feature: PaywallData.LocalizedConfiguration.Feature,
+    colors: TemplateConfiguration.Colors,
+) {
+    Row(
+        verticalAlignment = Alignment.Top,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+    ) {
+        feature.iconID?.let { PaywallIconName.fromValue(it) }?.let { icon ->
```

Comment:
Hmm is there a chance of some features having icons and some not? Asking since in that case, the rows text might not be aligned. This might be an ok edge case though.

Reply:
The frontend always asks for an icon. Not sure if we need to take care of that, but maybe we should have a backup icon just in case?

Reply:
If it's validated in the frontend, then I think it's fine to leave as is.